pkg/reconciler/artifactbuildrequest/controller.go

```diff
@@ -1,31 +1,18 @@
 package artifactbuildrequest
 
 import (
-	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/builder"
-	"sigs.k8s.io/controller-runtime/pkg/event"
-	"sigs.k8s.io/controller-runtime/pkg/predicate"
-
 	"github.com/redhat-appstudio/jvm-build-service/pkg/apis/jvmbuildservice/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
 func SetupNewReconcilerWithManager(mgr ctrl.Manager) error {
 	r := newReconciler(mgr)
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&v1alpha1.ArtifactBuildRequest{}, builder.WithPredicates(predicate.Funcs{
-			CreateFunc: func(e event.CreateEvent) bool {
-				return true
-			},
-			UpdateFunc: func(e event.UpdateEvent) bool {
-				return true
-			},
-			DeleteFunc: func(e event.DeleteEvent) bool {
-				//TODO possibly change to false over time
-				return true
-			},
-			GenericFunc: func(e event.GenericEvent) bool {
-				//TODO possibly change to false over time
-				return true
-			},
-		})).Complete(r)
+	return ctrl.NewControllerManagedBy(mgr).For(&v1alpha1.ArtifactBuildRequest{}).
+		//we can't use .Owns() here as controller is not true in the owner ref
+		Watches(&source.Kind{Type: &v1beta1.TaskRun{}}, &handler.EnqueueRequestForOwner{OwnerType: &v1alpha1.ArtifactBuildRequest{}, IsController: false}).
+		Watches(&source.Kind{Type: &v1alpha1.DependencyBuild{}}, &handler.EnqueueRequestForOwner{OwnerType: &v1alpha1.ArtifactBuildRequest{}, IsController: false}).
+		Complete(r)
```

Comment:
before reading the rest of this comment, to be clear, I like this ^^ more :-)

but to complete the loop on our conversation this morning with @sbose78 and the "shipwright" form for doing something similar that I have references a bunch of times recently, just so you are aware more precisely of what I'm referring to, where you are clearly well versed now in controller runtime nuance to compare/contrast, here it is:  https://github.com/shipwright-io/build/blob/main/pkg/reconciler/buildrun/controller.go 

take a peek as your curiosity deems worthy @stuartwdouglas